example/example.go

```diff
@@ -94,3 +141,97 @@ func main() {
 		log.Printf("Public ID: %v, URL: %v\n", asset.PublicID, asset.SecureURL)
 	}
 }
+
+// Generate a responsive srcset string for a given public ID by building URLs at multiple widths.
+func generateResponsiveSrcSet(cld *cloudinary.Cloudinary, publicID string) (string, error) {
+	widths := []int{200, 400, 800, 1200}
+	var parts []string
+
+	for _, w := range widths {
+		img, err := cld.Image(publicID)
+		if err != nil {
+			return "", fmt.Errorf("failed to initialize image %s: %w", publicID, err)
+		}
+		img.Transformation = fmt.Sprintf("c_scale,w_%d/f_auto/q_auto", w)
+
+		url, err := img.String()
+		if err != nil {
+			return "", fmt.Errorf("failed to build URL for width %d: %w", w, err)
+		}
+		parts = append(parts, fmt.Sprintf("%s %dw", url, w))
+	}
+
+	return strings.Join(parts, ", "), nil
+}
+
+// Upload a video with transformations applied on upload to generate posters or clips.
+func uploadVideoWithTransformations(cld *cloudinary.Cloudinary, ctx context.Context) *uploader.UploadResult {
+	uploadResult, err := cld.Upload.Upload(
+		ctx,
+		videoFilePath,
+		uploader.UploadParams{
+			PublicID:     "promo_clip",
+			Folder:       "videos/promos",
+			ResourceType: "video",
+			Eager:        "c_fill,h_360,w_640,b_black|c_crop,ar_16:9,e_volume:0.5,du_15",
+			Tags:         []string{"video", "promo"},
+		},
+	)
+	if err != nil {
+		log.Fatalf("failed to upload video: %v", err)
+	}
+
+	// uploadResult.SecureURL points to the original video; uploadResult.Eager to derivatives.
+	return uploadResult
+}
+
+// Delete a single asset by its Public ID.
+func deleteAsset(cld *cloudinary.Cloudinary, ctx context.Context, publicID string) {
+	_, err := cld.Upload.Destroy(
+		ctx,
+		uploader.DestroyParams{PublicID: publicID, ResourceType: "image"},
+	)
+	if err != nil {
+		log.Fatalf("failed to delete asset %s: %v", publicID, err)
+	}
+
+	// Asset deleted successfully.
+}
+
+// Bulk delete multiple assets by their Public IDs.
+func bulkDeleteAssets(cld *cloudinary.Cloudinary, ctx context.Context, publicIDs []string) {
+	resp, err := cld.Admin.DeleteAssets(
+		ctx,
+		admin.DeleteAssetsParams{PublicIDs: publicIDs},
+	)
+	if err != nil {
+		log.Fatalf("failed to bulk delete assets: %v", err)
+	}
+
+	// Print how many were deleted.
+	log.Printf("Deleted assets count: %d", len(resp.Deleted))
+}
+
+// List all assets in pages of up to maxResults, using cursor-based pagination.
+func listAssetsWithPagination(cld *cloudinary.Cloudinary, ctx context.Context, maxResults int) {
+	nextCursor := ""
+
+	for {
```

Comment:
This will list all assets in your cloud. On my cloud I have thousands of assets, it will run many api calls.

I would limit this to a few loops for demo purposes.